### DISCUSSION TITLE
MHCOZY 4 Channel Wireless ZigBee Relay Switch

### DISCUSSION THREAD

**Discussion by Plasmator-x64** [Koenkk/zigbee2mqtt#13785] [Devices]
Hello,

I'm trying to add the "MHCOZY 4 Channel Wireless ZigBee Relay Switch" to Z2M as a supported switch.

However, it shows up as "TuYa TS0004" - four (4) switches and a router with horrible performance.

I want to add the device with just four (4) switches and no router. Similar to the "Xiaomi WS-USC01" - mains connected switch with no router.

**Question - would an external converter allow me to do this ? ( add the device with just four (4) switches and no router )**

MHCOZY 4 Channel Wireless ZigBee Relay Switch
https://www.amazon.ca/gp/product/B091GS6J5W/

Thanks,
Plasmator

![71Q6XX3mBFL _AC_SL1200_](https://user-images.githubusercontent.com/94121123/187810251-776ea465-b533-4f81-9501-08c36aca47fa.jpg)

**Comment by Plasmator-x64** [comment]
Quick Update - Success !!!

I got the "MHCOZY 4 Channel Wireless ZigBee Relay Switch" working with an External Converter.

Now - each switch toggles the matching relay On/Off - even with the "Switch Type" set to "Momentary"

Still need some tweaks to complete the integration :-

1. Get Switches to function as Momentary
2. Confirm Routing is Disabled
3. Device Picture & Documentation
4. Official Support

[MHCOZY.js.txt](https://github.com/Koenkk/zigbee2mqtt/files/9474724/MHCOZY.js.txt)

![MHCOZY_00](https://user-images.githubusercontent.com/94121123/188055224-4016afa2-3180-4c71-bf14-e774107c044d.png)

![MHCOZY_01](https://user-images.githubusercontent.com/94121123/188055229-bf2c9d49-3f38-4d64-8c5c-a52f1594e4df.png)

![MHCOZY_02](https://user-images.githubusercontent.com/94121123/188055234-dab55964-1e3b-42f2-8b30-3eda866b7328.png)

**Reply by jerrm** [reply]
Does setting it as an endpoint in the converter really disable routing, or will it just confuse things like the map?

**Reply by andrei-lazarov** [reply]
It's not possible to change the router / non router type of a Zigbee device.
The converter only overwrites the text displayed in Z2M

**Comment by Plasmator-x64** [comment]
Yes - routing is now Disabled and no other EndDevice is connected to the Relay.

The ZigBee network map with the dedicated Routers are back to the original design.

I still need to tweak the switches so they function as Momentary and not toggle/latch

Not sure if this is possible ?
